Mathlib/NumberTheory/LSeries/HurwitzZetaValues.lean

```diff
@@ -0,0 +1,197 @@
+      ← Nat.cast_ofNat (R := ℂ), ← Nat.cast_mul, cpow_natCast, ofReal_pow, ofReal_natCast]
+  · simp only [ofReal_mul, ofReal_div, ofReal_pow, ofReal_natCast, ofReal_ofNat,
+      ofReal_neg, ofReal_one]
+    congr 1
+    have : (Polynomial.bernoulli (2 * k)).map (algebraMap ℚ ℂ) = _ :=
+      (Polynomial.map_map (algebraMap ℚ ℝ) ofReal _).symm
+    rw [this, ← ofReal_eq_coe, ← ofReal_eq_coe]
+    apply Polynomial.map_aeval_eq_aeval_map
+    simp only [Algebra.id.map_eq_id, RingHomCompTriple.comp_eq]
+  · rw [← Nat.cast_ofNat, ← Nat.cast_one, ← Nat.cast_mul, natCast_re, Nat.cast_lt]
+    omega
+
+/--
+Express the value of `sinZeta` at an odd integer `> 1` as a value of the Bernoulli polynomial.
+
+Note that this formula is also correct for `k = 0` (i.e. for the value at `s = 1`), but we do not
+prove it in this case, owing to the additional difficulty of working with series that are only
+conditionally convergent.
+-/
+theorem sinZeta_two_mul_nat_add_one (hk : k ≠ 0) (hx : x ∈ Icc (0 : ℝ) 1) :
+    sinZeta x (2 * k + 1) = (-1) ^ (k + 1) * (2 * π) ^ (2 * k + 1) / 2 / (2 * k + 1)! *
+      ((Polynomial.bernoulli (2 * k + 1)).map (algebraMap ℚ ℂ)).eval ↑x := by
+  rw [← (hasSum_nat_sinZeta x (?_ : 1 < re (2 * k + 1))).tsum_eq]
+  refine Eq.trans ?_ <| (congr_arg ofReal' (hasSum_one_div_nat_pow_mul_sin hk hx).tsum_eq).trans ?_
+  · rw [ofReal_tsum]
+    refine tsum_congr fun n ↦ ?_
+    rw [mul_comm (1 / _), mul_one_div, ofReal_div, mul_assoc (2 * π), mul_comm x n, ← mul_assoc]
+    congr 1
+    rw [← Nat.cast_ofNat, ← Nat.cast_mul, ← Nat.cast_add_one, cpow_natCast, ofReal_pow,
+      ofReal_natCast]
+  · simp only [ofReal_mul, ofReal_div, ofReal_pow, ofReal_natCast, ofReal_ofNat,
+      ofReal_neg, ofReal_one]
+    congr 1
+    have : (Polynomial.bernoulli (2 * k + 1)).map (algebraMap ℚ ℂ) = _ :=
+      (Polynomial.map_map (algebraMap ℚ ℝ) ofReal _).symm
+    rw [this, ← ofReal_eq_coe, ← ofReal_eq_coe]
+    apply Polynomial.map_aeval_eq_aeval_map
+    simp only [Algebra.id.map_eq_id, RingHomCompTriple.comp_eq]
+  · rw [← Nat.cast_ofNat, ← Nat.cast_one, ← Nat.cast_mul, ← Nat.cast_add_one, natCast_re,
+      Nat.cast_lt, lt_add_iff_pos_left]
+    exact mul_pos two_pos (Nat.pos_of_ne_zero hk)
+
+/-- Reformulation of `cosZeta_two_mul_nat` using `Gammaℂ`. -/
+theorem cosZeta_two_mul_nat' (hk : k ≠ 0) (hx : x ∈ Icc (0 : ℝ) 1) :
+    cosZeta x (2 * k) = (-1) ^ (k + 1) / (2 * k) / Gammaℂ (2 * k) *
+      ((Polynomial.bernoulli (2 * k)).map (algebraMap ℚ ℂ)).eval ↑x := by
+  rw [cosZeta_two_mul_nat hk hx]
+  congr 1
+  have : (2 * k)! = (2 * k) * Complex.Gamma (2 * k) := by
+    rw [(by { norm_cast; omega } : 2 * (k : ℂ) = ↑(2 * k - 1) + 1), Complex.Gamma_nat_eq_factorial,
+      ← Nat.cast_add_one, ← Nat.cast_mul, ← Nat.factorial_succ, Nat.sub_add_cancel (by omega)]
+  simp_rw [this, Gammaℂ, cpow_neg, ← div_div, div_inv_eq_mul, div_mul_eq_mul_div, div_div]
+  congr 2
+  · rw [(by simp : 2 * (k : ℂ) = ↑(2 * k)), cpow_natCast]
+  · ring
```

Comment:
```suggestion
  norm_num
  rw [(by simp : 2 * (k : ℂ) = ↑(2 * k)), cpow_natCast]
  ring
```

Reply:
I tried applying this change, but it didn't compile. Maybe there was a typo? Could you take another look and let me know what you intended here?

Reply:
I am sorry, it was a bad copy-pasting. The following compiles for me:

```suggestion
  norm_cast
  ring_nf
```

Reply:
Thanks, that works indeed!